impeller/entity/shaders/linear_gradient_fill.frag

```diff
@@ -25,11 +26,17 @@ void main() {
     gradient_info.end_point - gradient_info.start_point
   );
   float t = dot / (len * len);
-  frag_color = IPSampleWithTileMode(
+  if ((t < 0.0 || t >= 1.0) && gradient_info.tile_mode == kTileModeDecal) {
```

Comment:
is this an issue we should fix in IPSampleWithTileMode?

Reply:
This is specifically a problem for linear sampling, so maybe we should have `IPSampleLinear` and `IPSampleLinearWithTileMode`, which is basically the same but with an extra `half_texel` param? We need to make this fix almost any time we do linear sampling (as @flar pointed out a couple of weeks back), so these utilities will probably get a good amount of use.

Reply:
Done. `IPSampleLinear` and `IPSampleLinearWithTileMode ` have been added.